nfs.tf

```diff
@@ -1,12 +1,48 @@
 resource "openstack_compute_instance_v2" "nfs-server" {
   name            = "${var.name_prefix}nfs${var.name_suffix}"
+  image_id        = "${data.openstack_images_image_v2.vgcn-image.id}"
   flavor_name     = "${var.flavors[1]}"
-  image_name      = "${var.image}"
   key_pair        = "${openstack_compute_keypair_v2.my-cloud-key.name}"
   security_groups = "${var.secgroups}"
   network         = "${var.network}"
 
-  user_data = <<-EOF
+  block_device {
+    uuid                  = "${openstack_blockstorage_volume_v2.volume_nfs_data.id}"
+    source_type           = "image"
+    destination_type      = "local"
+    boot_index            = 0
+    delete_on_termination = true
+  }
+
+  block_device {   
+    uuid                  = "${openstack_blockstorage_volume_v2.volume_nfs_data.id}"
+    source_type           = "volume"
+    destination_type      = "volume"
+    boot_index            = -1
+    delete_on_termination = true
+  }
+
+  user_data = "${data.template_cloudinit_config.nfs-share.rendered}"
+}
+
+
+resource "openstack_blockstorage_volume_v2" "volume_nfs_data" {
+  name = "${var.name_prefix}volume_nfs_data"
+  size = "${var.nfs_disk_size}"
+}
+
+data "template_cloudinit_config" "nfs-share" {
+  gzip          = true
+  base64_encode = true
+
+  part {
```

Comment:
Any reason you've used parts here, and not write_files + runcmd?

Reply:
Also: does this only get run once? I'm not sure what cloud-init portions are run just once or during every boot.

Reply:
Is not clear to me why, but the disk_setup and fs_setup directives don't work in this setup.
I have rebooted the vm several times and the whole cloud-init config has been applied just once